### ISSUE TITLE
CRD Rule format error:  type conversion failed (kind: NvClusterSecurityRule)

### ISSUE THREAD

**Issue by pbloed** [neuvector/neuvector#1283]
**Environment**
Platform: Openshift 4.13.34/kubernetes 1.26/Neuvector 5.3.1/Chart 2.7.4

**Describe the bug**
When a NvClusterSecurityRule object is created in the neuvector namespace the NvClusterSecurityRule is not loaded in Neuvector.
After pod restart the log shows above type conversion error and the object is removed!

**To Reproduce**
1. Create demo.yaml
```yaml
apiVersion: neuvector.com/v1
kind: NvClusterSecurityRule
metadata:
  name: demo
  namespace: ""
spec:
  egress: null
  file: null
  ingress: null
  process: null
  target:
    selector:
      criteria:
      - key: address
        op: =
        value: 0.0.0.0/16
      name: world
    policymode: Monitor
```
2. Create kubernetes object
```bash
oc create -n neuvector -f demo.yaml
```

3. Login Neuvector UI en check if network rule is created

4. Restart controller pod and check log for "CrossCheckCrd" error(s)

**Expected behavior**
NvClusterSecurityRule is parsed correctly without a controller restart and Network Rule is added at runtime.

**Logging**
```
2024-04-08T11:53:02.362|INFO|CTL|rest.(*nvCrdHandler).parseCrdContent: CRD validate fail :   CRD Rule format error:  type conversion failed (kind: NvClusterSecurityRule)
2024-04-08T11:53:02.362|ERRO|CTL|rest.CrossCheckCrd: - error=  CRD Rule format error:  type conversion failed (kind: NvClusterSecurityRule) name=demo
```

**Comment by williamlin-suse** [comment]
ip group doesn't support policymode.
you can comment out the line "policymode: Monitor" and try again

**Comment by pbloed** [comment]
Removed the line and same result.

```yaml
apiVersion: neuvector.com/v1
kind: NvClusterSecurityRule
metadata:
  name: demo
spec:
  target:
    selector:
      criteria:
      - key: address
        op: =
        value: 0.0.0.0/16
      name: world
```

```
2024-04-08T14:13:05.033|INFO|CTL|rest.CrossCheckCrd: - kvCrdKind=NvSecurityRule kvOnly=false len(objs)=1 len(recordList)=0 rscType=nvclustersecurityrules
2024-04-08T14:13:05.046|INFO|CTL|rest.(*nvCrdHandler).parseCrdContent: CRD validate fail :   CRD Rule format error:  type conversion failed (kind: NvClusterSecurityRule)
```

**Comment by skuijper** [comment]
I'm seeing the same issue. It seems like the CRD API validation does not match the "CrossCheckCrd" validation in the controller.

**Comment by becitsthere** [comment]
This will be resolved in 5.3.2 release, which we will release soon.

**Comment by pbloed** [comment]
The "CrossCheckCrd" validation works in 5.3.2 but a controller restart is still required before clusterpolicyrules are added in Neuvector.
